shared/metrics/src/main/java/io/pravega/shared/metrics/DynamicLoggerImpl.java

```diff
@@ -21,20 +21,27 @@
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 
+import io.pravega.shared.MetricsNames;
 import lombok.extern.slf4j.Slf4j;
 
+import static io.pravega.shared.MetricsNames.metricKey;
+import static io.pravega.shared.MetricsNames.COUNTER_SUFFIX;
+import static io.pravega.shared.MetricsNames.GAUGE_SUFFIX;
+import static io.pravega.shared.MetricsNames.METER_SUFFIX;
+
 @Slf4j
 public class DynamicLoggerImpl implements DynamicLogger {
+
     private final long cacheSize;
     private final long cacheEvictionDuration;
 
-    private final MetricRegistry metrics;
+    private final MeterRegistry metrics;
     private final StatsLogger underlying;
```

Comment:
I feel that `underlying` is lacking something, like `underlyingStatsLogger` or `backendLogger`. Otherwise is not clear when it is use throughout the code.

Reply:
Are you talking about enhancement beyond this migration? Not quite getting your point.

Reply:
Sorry, I just referred to the name of the field `underlying` is ambiguous and could be renamed to make it clearer.